mat/dense.go

```diff
@@ -201,10 +201,16 @@ func (m *Dense) T() Matrix {
 	return Transpose{m}
 }
 
-// ColView returns a VecDense reflecting the column j, backed by the matrix data.
+// ColView returns a Vector reflecting the column j, backed by the matrix data.
 //
 // See ColViewer for more information.
-func (m *Dense) ColView(j int) *VecDense {
+func (m *Dense) ColView(j int) Vector {
+	return m.ColViewVecDense(j)
+}
+
+// ColViewVecDense returns a VecDense reflecting the column j, backed by
+// the matrix data.
+func (m *Dense) ColViewVecDense(j int) *VecDense {
```

Comment:
On reflection this signature should be intverted; `func (v *VecDense) ColViewOf(m *Dense, j int)`. The same is true for the row equivalent.

By doing this we gain many allocation reductions cases where vector views are taken by getting the destination vector prior to a loop (possible getting it from a `*VecDense` pool).

The signature that exists is a product of history and us not thinking about changes when they happened; the views originally were slices, which meant that they were not allocated on the heap when a view was taken.